controller/api/models.py

```diff
@@ -353,6 +367,58 @@ def _start_containers(self, to_add):
         if set([c.state for c in to_add]) != set(['up']):
             err = 'warning, some containers failed to start'
             log_event(self, err, logging.WARNING)
+        # if the user specified a health check, try checking to see if it's running
+        try:
+            config = self.config_set.latest()
+            if 'HEALTHCHECK_URL' in config.values.keys():
+                self._healthcheck(to_add, config.values)
+        except Config.DoesNotExist:
+            pass
+
+    def _healthcheck(self, containers, config):
+        # if at first it fails, back off and try again at 10%, 50% and 100% of INITIAL_DELAY
+        intervals = [1.0, 0.1, 0.5, 1.0]
+        # HACK (bacongobbler): we need to wait until publisher has a chance to publish each
```

Comment:
Instead of hacking a `time.sleep()`, could we perhaps just use the etcd client to watch for the service to be registered, and have a timeout at 20-30 seconds?

With this, all deploys with healthchecks will add at least 20s of time to deploy. Even if the container is up and running, and passes health checks after 3-4 seconds.